### ISSUE TITLE
Regression: can't disable a legend in a specific layer in >=4.8.0

### ISSUE THREAD

**Issue by dechamps** [vega/vega-lite#6259]
Consider the following:

```json
{
  "layer": [{
    "mark": { "type": "line" },
    "encoding": {
      "color": {
        "type": "nominal",
        "field": "c",
        "legend": {}
      }
    }
  }, {
    "mark": { "type": "circle" },
    "encoding": {
      "color": {
        "type": "nominal",
        "field": "c",
        "legend": null
      }
    }
  }],
  "data": {"values": [{"c": "A"}, {"c": "B"}]}
}
```

Notice in particular the fact that the `legend` field is non-null in the first layer, but null in the second layer. The intent is to generate the legend from the first layer only (in terms of scale, symbol, etc.), and ignore the second layer.

In vega-lite **4.7.0** this [produces](https://jsfiddle.net/2rjysedn/) the following, which is what I want:

![visualization(48)](https://user-images.githubusercontent.com/866859/78450109-24912f80-7674-11ea-9a7e-3807119c45d6.png)

But starting from **4.8.0** the [output](https://jsfiddle.net/2rjysedn/2/) changed, and the `circle` layer is not ignored anymore:

![visualization(49)](https://user-images.githubusercontent.com/866859/78450147-6621da80-7674-11ea-978d-c66b6680ddb0.png)

In [Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEA2CGCeBTATpAXGA2qCFIFtZkBrdMYKAF3gAdEyYBLAO3rAF8AacXSRZgMYB7ACYsA5mRy48w6ENQZpMvNToNmQ-C1jRI3FbwBmjRNBEMB+noZiJx-C0vY3cLme46dyrgkVJKVLT0GJACjMgC0GxcvvzCYsySSq6yQvKKPoaqwRpaOnoG2ZAmZk5QVkW20Q7M5cwArtDQqRyunuwAukWQIrCUsFKQAG66DYgAzmTYYQwAgpBc5LOhAEKLnS4uQA) the log includes the following interesting warning:

```
[Warning] Conflicting legend property "disable" (false and true). Using false.
```

Now, in this simple case, `"resolve": {"legend": {"color": "independent"}}` [can be used as a workaround](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEA2CGCeBTATpAXGA2qCFIFtZkBrdMYKAF3gAdEyYBLAO3rAF8AacXSRZgMYB7ACYsA5mRy48w6ENQZpMvNToNmQ-C1jRI3FbwBmjRNBEMB+noZiJx-C0vY3cLme46dyrgkVJKVLT0GJACjMgC0GxcvvzCYsySSq6yQvKKPoaqwRpaOnoG2ZAmZk5QVkW20Q7M5cwArtDQqRyunuwAukWQyIgAzukAbiHkdrXlwGHpCgwsIoh0dfyUkOyxeCKwlLBSkEO6DQNk2GEMAIJr3lNWoQBCa50uLkA). The problem is, that only works for simple cases. If I [facet the chart](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1STkhR4yb3d0sGBpcI0S5PLjWSLSqFMLKxooyVgOXVhcyCcsgLcPL0RfMOzA+GDQmpHCG2iWdi5eAVqNYhFxFqWpqFz8xELEAFceHmWnWsqygF0ehV0yRTHoWeodgrVIIJDJLjEyRj33BhpBczhBpNd5JAxOgUO9oLwDvcDNQiJIAIwCQKSACCMkEKMkACZMR9JAAhGSXEolIA) for example, there doesn't seem to be a way to indicate that I want the legends to be independent between *layers* but not between *facets* (see #6261).

Another workaround to solve the issue in this simple case is `"legend": {"symbolType": "stroke"}`, which [forces the legend to use the symbol I want](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBuDGD2A7WBDALpAXGA2qCEkAZirAKYbbCRoCeADmVlIvALYCWiKANpADRQiHMjwAmzYpAC+-cAUgBnRrGb4ChHilpkATszzyNhNil0BrNVDqNJPLkzCyjxyGWTwxXAOZqXxqAQeeH0qfwDrBiZsSFZObj45CNdhUQkY1STkhR4yb3d0sGpFWjYAI3geABUoyUU0XXhzJmlwjVbkjuNZIraoUwsrGlqMjl1YXMgnLIC3Dy9EXzDswMqQvxXCG2iWdi5eAT6NYhFxSVUjpz6u9oBdFy7pW6zIMXQUNRheAFcyRQNqERJABGASBSQAQRkgkBkgATGDIKoYgAhGS3VqtIA). But sadly, this too does not work in all cases: for example this technique can't be used to solve #6258.

**Comment by domoritz** [comment]
To achieve what you want, you can use resolve to make the legend (or scale) independent.

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "layer": [
    {
      "mark": {"type": "line"},
      "encoding": {"color": {"type": "nominal", "field": "c"}}
    },
    {
      "mark": {"type": "circle"},
      "encoding": {"color": {"type": "nominal", "field": "c", "legend": null}}
    }
  ],
  "data": {"values": [{"c": "A"}, {"c": "B"}]},
  "resolve": {
    "legend": {"color": "independent"}
  }
}
```

The previous behavior was not working as we intended so it's not a regression., I wonder whether we should just ignore `legend: {}` instead of treating it as though you want a legend.

**Comment by dechamps** [comment]
> To achieve what you want, you can use resolve to make the legend (or scale) independent.

As I mentioned in my original post, I already tried that. That only gives me the behaviour I want on non-faceted plots. If I try it on a faceted plot, I run directly into #6261.

I understand that the 4.7.0 behaviour was not intended, but that doesn't change the fact that I used to be able to make Vega-Lite ignore the legend in a specific layer of a faceted chart in 4.7.0. In 4.8.0 it doesn't look like there's any way to do that any more, with no workaround available. I ended up freezing my dependency on Vega-Lite to 4.7.0 because I couldn't find any other way.

For me to be able to depend on 4.8.0 and later, either this needs to be fixed, or #6261 needs to be fixed, or #6258 needs to be fixed. Otherwise I'm stuck between a rock and a hard place :(

**Comment by domoritz** [comment]
Okay, let's take a step back and see how we can get a solution that works for everything. I definitely want you to be able to update! Going to your original request, you want "The intent is to generate the legend from the first layer only (in terms of scale, symbol, etc.), and ignore the second layer." Fundamentally, if the scale is shared between the two layers, there is no way to use only the scale of one layer. Do you agree?

If you do, then do I understand that you want the legend to be defined by the mark in only one layer? So is what you want only the right mark? If so, we can see how we can achieve that. 

The underlying issue is that Vega-Lite needs to resolve properties from different views to a single property. `disable: true` and `disable: false` in your example and that results in the warning `[Warning] Conflicting legend property "disable" (false and true). Using false.`. The mark property is then taken from the top-most layer.

We could change some of the resolution logic here but I hope you understand why I don't think the behavior in 4.8 is wrong. 

https://github.com/vega/vega-lite/issues/6258 looks like a bug so let me take a look at that and if that fixes your issue, then that would be great.

**Comment by dechamps** [comment]
> Fundamentally, if the scale is shared between the two layers, there is no way to use only the scale of one layer. Do you agree?

Yeah, I think I used the wrong terminology here - what I meant by "scale" was actually things like "legend type" or any other [legend property](https://vega.github.io/vega-lite/docs/legend.html#legend-properties), really. Basically I want Vega-Lite to only use the legend properties from one layer, and act as if the second layer does not exist for the purposes of generating the legend.

(To give a bit more context about why I want to do this: I have a main, fully visible layer with lines, and I have another transparent circle layer that is only there for interactivity/selection/tooltip purposes. Because the transparent layer is only an "implementation detail" of the chart, its legend is completely irrelevant and should not affect the legend of the combined chart.)

Don't get me wrong, I do agree that the behaviour described in my original post could be construed as a "hack". And I do agree that `"resolve": {"legend": {"color": "independent"}}` should be the correct, clean, obvious way to fix this. And I would be happy to use it, if it weren't for #6261 :(

**Comment by domoritz** [comment]
Thanks for your understanding. The dilemma for me is that I don't know what a consistent behavior should be to achieve what you want in your original example in this thread. 

If you have an idea, we can change this issue from a bug into a feature request and tweak the resolution behavior between layers for guides (not just legends but also axes). 

Otherwise, you may need to wait for a fix for #6261. If you have free cycles, maybe you can take a stab at a pull request. We are very open to contributions.
